### PR TITLE
Change language statistics to save size instead of percentage (#11681)

### PR DESCRIPTION
**Pull request by lafriks**
Backport #11681 & #11692

### CODE REVIEW

**Comment by lafriks** [comment]
More fixes are coming

File: models/repo_language_stats.go [hunk 1]
```diff
@@ -20,11 +20,28 @@ type LanguageStat struct {
 	CommitID    string
 	IsPrimary   bool
 	Language    string             `xorm:"VARCHAR(30) UNIQUE(s) INDEX NOT NULL"`
-	Percentage  float32            `xorm:"NUMERIC(5,2) NOT NULL DEFAULT 0"`
+	Percentage  float32            `xorm:"-"`
+	Size        int64              `xorm:"NOT NULL DEFAULT 0"`
```
Comment (lunny) [inline models/repo_language_stats.go:24]: `size` should have `index` tag on `xorm` if you want to `order by` with it.

Comment (lafriks) [reply models/repo_language_stats.go:24]: Results will be already filtered to only one repo so there will be less 10 mostly so that should not be problem

**Comment by CirnoT** [comment]
Please integrate https://github.com/go-gitea/gitea/pull/11692

**Comment by lafriks** [comment]
@CirnoT done

**Review by CirnoT — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by lafriks** [comment]
Make lgtm work

**Pull request merged**